src/DssSpell.t.sol

```diff
@@ -974,48 +1049,257 @@ contract DssSpellTest is DssSpellTestBase {
+        assertEq(ink, 1_000_000_000_000 * WAD);
+        assertEq(art, 0);
+
+        assertEq(AllocatorRegistryLike(ALLOCATOR_REGISTRY).buffers(ALLOCATOR_ILK), ALLOCATOR_SPARK_BUFFER);
+        assertEq(address(AllocatorVaultLike(ALLOCATOR_SPARK_VAULT).jug()), address(jug));
+
+        assertEq(usds.allowance(ALLOCATOR_SPARK_BUFFER, ALLOCATOR_SPARK_VAULT), type(uint256).max);
+
+        assertEq(AllocatorRolesLike(ALLOCATOR_ROLES).ilkAdmins(ALLOCATOR_ILK), ALLOCATOR_SPARK_PROXY);
+
+        assertEq(AllocatorVaultLike(ALLOCATOR_SPARK_VAULT).wards(pauseProxy),  0);
+        assertEq(AllocatorVaultLike(ALLOCATOR_SPARK_VAULT).wards(ALLOCATOR_SPARK_PROXY), 1);
+
+        assertEq(AuthLike(ALLOCATOR_SPARK_BUFFER).wards(pauseProxy),  0);
+        assertEq(AuthLike(ALLOCATOR_SPARK_BUFFER).wards(ALLOCATOR_SPARK_PROXY), 1);
+
+        assertEq(reg.count(),               previousIlkRegistryCount + 1);
+        assertEq(reg.pos(ALLOCATOR_ILK),    previousIlkRegistryCount);
+        assertEq(reg.join(ALLOCATOR_ILK),   address(0));
+        assertEq(reg.gem(ALLOCATOR_ILK),    address(0));
+        assertEq(reg.dec(ALLOCATOR_ILK),    0);
+        assertEq(reg.class(ALLOCATOR_ILK),  5);
+        assertEq(reg.pip(ALLOCATOR_ILK),    PIP_ALLOCATOR_SPARK);
+        assertEq(reg.xlip(ALLOCATOR_ILK),   address(0));
+        assertEq(reg.name(ALLOCATOR_ILK),   string("ALLOCATOR-SPARK-A"));
+        assertEq(reg.symbol(ALLOCATOR_ILK), string("ALLOCATOR-SPARK-A"));
+
+        // Draw & Wipe from Vault
+        vm.prank(address(ALLOCATOR_SPARK_PROXY));
+        AllocatorVaultLike(ALLOCATOR_SPARK_VAULT).draw(1_000 * WAD);
+        assertEq(usds.balanceOf(ALLOCATOR_SPARK_BUFFER), 1_000 * WAD);
+
+        vm.prank(address(ALLOCATOR_SPARK_PROXY));
+        AllocatorVaultLike(ALLOCATOR_SPARK_VAULT).wipe(1_000 * WAD);
+        assertEq(usds.balanceOf(ALLOCATOR_SPARK_BUFFER), 0);
+    }
+
+    function testsWhitelistSparkProxyOnLitePsm() public {
+         _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "TestError/spell-not-done");
+
+        assertEq(LitePsmLike(LITE_PSM).bud(ALLOCATOR_SPARK_PROXY), 1);
+    }
+
+    function testMedianValidators() public {
+         _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "TestError/spell-not-done");
+
+        address[] memory validators = new address[](4);
+        validators[0] = ETH_GLOBAL_VALIDATOR;
+        validators[1] = MANTLE_VALIDATOR;
+        validators[2] = NETHERMIND_VALIDATOR;
+        validators[3] = EULER_VALIDATOR;
+
+        for (uint i = 0; i < validators.length; i++) {
+            assertEq(MedianLike(BTC_USD_MEDIAN).orcl(validators[1]), 1);
+            assertEq(MedianLike(ETH_USD_MEDIAN).orcl(validators[1]), 1);
+            assertEq(MedianLike(WSTETH_USD_MEDIAN).orcl(validators[1]), 1);
+            assertEq(MedianLike(MKR_USD_MEDIAN).orcl(validators[1]), 1);
```

Comment:
Why not validate all other oracles here with a nested loop?
Also you probably want to check the state **before** the execution of the spell.